lib/vrl/stdlib/src/encode_logfmt.rs

```diff
@@ -0,0 +1,234 @@
+                    output.write_char(' ').map_err(|_| "write error")?;
+                }
+                encode_field(&mut output, field, val).map_err(|_| "write error")?;
+                seen_fields.insert(field);
+            }
+        }
+
+        for (idx, (key, value)) in input.iter().enumerate() {
+            if seen_fields.contains(key) {
+                continue;
+            }
+
+            if idx > 0 || !seen_fields.is_empty() {
+                output.write_char(' ').map_err(|_| "write error")?;
+            }
+            encode_field(&mut output, key, value).map_err(|_| "write error")?;
+        }
+
+        Ok(output)
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct EncodeLogfmt;
+
+impl Function for EncodeLogfmt {
+    fn identifier(&self) -> &'static str {
+        "encode_logfmt"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::OBJECT,
+                required: true,
+            },
+            Parameter {
+                keyword: "fields_ordering",
+                kind: kind::ARRAY,
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        let fields = arguments.optional("fields_ordering");
+
+        Ok(Box::new(EncodeLogfmtFn { value, fields }))
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "encode object",
+                source: r#"encode_logfmt!({"lvl": "info", "msg": "This is a message", "log_id": 12345})"#,
+                result: Ok(r#"s'log_id=12345 lvl=info msg="This is a message"'"#),
+            },
+            Example {
+                title: "encode object with fields ordering",
+                source: r#"encode_logfmt!({"msg": "This is a message", "lvl": "info", "log_id": 12345}, ["lvl", "msg"])"#,
+                result: Ok(r#"s'lvl=info msg="This is a message" log_id=12345'"#),
+            },
+        ]
+    }
+}
+
+#[derive(Clone, Debug)]
+struct EncodeLogfmtFn {
+    value: Box<dyn Expression>,
+    fields: Option<Box<dyn Expression>>,
+}
+
+fn resolve_fields(fields: &Value) -> Vec<String> {
+    match fields {
+        Value::Array(arr) => arr
+            .iter()
+            .filter_map(|v| match v {
+                Value::Bytes(bytes) => Some(String::from_utf8_lossy(&bytes)),
+                _ => None,
+            })
+            .map(Into::into)
+            .collect(),
+        _ => vec![],
+    }
+}
+
+impl Expression for EncodeLogfmtFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let value = self.value.resolve(ctx)?;
+        let fields = match &self.fields {
+            None => vec![],
+            Some(expr) => match expr.resolve(ctx) {
+                Ok(val) => resolve_fields(&val),
+                Err(_) => vec![],
```

Comment:
I think we should bubble up this error the way you did when resolving the value on 148.